code/game/gamemodes/antag_datum.dm

```diff
@@ -0,0 +1,197 @@
+/mob/living
+	var/list/antag_datums
+
+/mob/living/proc/gain_antag_datum(datum_type)
+	if(!islist(antag_datums))
+		antag_datums = list()
+	var/datum/antagonist/D = new datum_type()
+	. = D.give_to_body(src)
+
+/mob/living/proc/has_antag_datum(type, check_subtypes)
+	if(!islist(antag_datums))
+		return FALSE
+	for(var/i in antag_datums)
+		var/datum/antagonist/D = i
+		if(check_subtypes)
+			if(istype(D, type))
+				return D
+		else
+			if(D.type == type)
+				return D
+	return FALSE
+
+/datum/antagonist
+	var/mob/living/owner
+	var/some_flufftext = "yer an antag larry"
+
+/datum/antagonist/Destroy()
+	owner = null
+	return ..()
+
+/datum/antagonist/proc/can_be_owned(mob/living/new_body)
+	return new_body && !new_body.has_antag_datum(type, TRUE)
+
+/datum/antagonist/proc/give_to_body(mob/living/new_body) //tries to give an antag datum to a mob. cancels out if it can't be owned by the new body
+	if(new_body && can_be_owned(new_body))
+		new_body.antag_datums += src
+		owner = new_body
+		on_gain()
+		. = src //return the datum if successful
+	else
+		qdel(src)
+		. = FALSE
+
+/datum/antagonist/proc/on_gain() //on initial gain of antag datum, do this. should only be called once per datum
+	apply_innate_effects()
+	if(some_flufftext)
+		owner << some_flufftext
+
+/datum/antagonist/proc/apply_innate_effects() //applies innate effects to the owner, may be called multiple times due to mind transferral
+	//antag huds would go here if antag huds were less completely unworkable as-is
+
+/datum/antagonist/proc/remove_innate_effects() //removes innate effects from the owner, may be called multiple times due to mind transferral
```

Comment:
Todo note that it should only be called once per mob though in most cases multiple calls shouldn't matter